src/api/nodeGrouping/types.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright Linkurious SAS 2012 - 2024
+ *
+ * - Created on 2024-02-06.
+ */
+
+export interface NodeGroupingRule {
+  id: number;
+  sourceKey: string;
+  name: string;
+  groupingType: NodeGroupingType;
+  groupingOptions: NodeGroupingOptions[NodeGroupingType];
+  canDelete: boolean;
+}
+
+export enum NodeGroupingType {
+  PROPERTY_KEY = 'propertyKey'
+}
+
+export type NodeGroupingOptions = {
+  [NodeGroupingType.PROPERTY_KEY]: PropertyKeyNodeGroupingOptions;
+};
+
+export interface PropertyKeyNodeGroupingOptions {
+  itemType: string;
+  propertyKey: string;
+}
+
+export interface CreateNodeGroupingRuleParams extends Pick<NodeGroupingRule, 'sourceKey' | 'name'> {
```

Comment:
@mzorkany I think we should extend IDataSourceParams where sourceKey is optional, which allows the frontend to call that API without sending each time the source key.
 The server gets the sourcekey value from the url